### ISSUE TITLE
dataset class

### ISSUE THREAD

**Issue by jamesrkg** [Quantipy/quantipy#27]
Can we wrap meta and data into a new class, suggested name 'dataset', which simply has the existing data and meta as attributes, and can benefit from a slew of class methods capable of treating the meta and the data as two sides of the same coin?

```python

class Dataset():
	"""
	Container for treating Quantipy meta and data as a unified whole.

	A Dataset provides a pathway for intrinsic methods capable of 
	editing meta and data by leveraging everything known both.
	"""

	self.meta = None
	self.data = None

	def __init__(self, meta=None, data=None):
		if not meta is None and not data is None:
			self.meta = meta
			self.data = data

	def load(self, path_json, path_csv):
		self.meta = load_json('%s.json'.format(path_name))
		self.data = pd.DataFrame.from_csv(path_csv)
```

There are a practically endless number of uses for a class like this. Some basic examples:

```python
qpd = qp.Dataset(meta, data)

qpd.variables() 				# return a list of all known variables
qpd.variables('int') 			# return a list of all int variables

qpd.add_variable(...)			# add a new variable (meta and data)
qpd.del_variable(...)			# delete a variable (meta and data)

qpd.save(path_json, path_csv)	# save meta and data to disk (as 2 files)
qpd.load(path_json, path_csv)	# load meta and data from disk (as 2 files)
```

And so on...

This class would provide space for a lot of currently free-form functions to become class methods.

For example:

```python
data['QMP_radio_stations_xb'] = recode(
    meta, data,
    target='QMP_radio_stations_xb',
    mapper={
        901: {'radio_stations': frange('1-23, 92, 94, 141')}
    }, 
    append=False
)
```

Could become the much more intuitive:

```python
qpd['QMP_radio_stations_xb'].recode(
    mapper={
        901: {'radio_stations': frange('1-23, 92, 94, 141')}
    }, 
    append=False
)
```

**Comment by jamesrkg** [comment]
To be clear about what I mean, I don't think this class needs to ever be serialized. The save/load methods should read/write the existing json/csv files we already have. This is purely a wrapper providing a framework for class methods.

**Comment by alextanski** [comment]
A first version of this will include:
`.describe(): `to look at meta
`.filter()`: to filter files inplace or create new DataSets
`.set_missings()`: missing by meta, see #445 
.`read_FORMAT()`: to read data files
`.split()`: to return data and meta components

**Comment by alextanski** [comment]
I am redirecting the issues #327, #432, #445 into this issue to get a basis for further dev:
Dev is going on in the core lib, branch: **i27-DataSet_class_to_core_lib**

**Comment by jamesrkg** [comment]
Resolved by #463.
